### PR TITLE
Fix (Revit): fix for nested geometry instances

### PR DESCRIPTION
**Pull request by connorivy**
## Changes:

- if an element is being converted as an instance, then first try to get the mesh from a geometry instance object on the top level of the Revit element's GeometryObject 
- add logging to make sure that our assumptions about the way element's GeometryElements are structured

## Screenshots:

Before this PR
image

After this PR
image

## Validation of changes:

All tested element's look the same in the viewer as in Revit

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs [hunk 1]
```diff
@@ -127,6 +85,169 @@ void SortGeometry(GeometryElement geom, Transform inverseTransform = null)
...
+      Transform inverseTransform = null
+    )
+    {
+      foreach (GeometryObject geomObj in geom)
+      {
+        switch (geomObj)
+        {
+          case Solid solid:
+            // skip invalid solid
+            if (solid.Faces.Size == 0 || Math.Abs(solid.SurfaceArea) == 0) continue;
```
Comment (jozseflkiss) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs:224]: might use `Math.Abs(solid.SurfaceArea) < Eps` if it is a floating point number

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs [hunk 2]
```diff
@@ -127,6 +85,169 @@ void SortGeometry(GeometryElement geom, Transform inverseTransform = null)
...
+      }
+      else if (hasSymbolGeom)
+      {
+        if (topLevelSolidsCount > 0)
+        {
+          SpeckleLog.Logger.Warning("Element of type {elementType} with uniqueId {uniqueId} has valid symbol geometry and {numSolids} top level solids. See comment on method SortInstanceGeometry for link to RevitAPI docs that leads us to believe this shouldn't happen", element.GetType(), element.UniqueId, topLevelSolidsCount);
+        }
+        if (topLevelMeshesCount > 0)
+        {
+          SpeckleLog.Logger.Warning("Element of type {elementType} with uniqueId {uniqueId} has valid symbol geometry and {numSolids} top level meshes. See comment on method SortInstanceGeometry for link to RevitAPI docs that leads us to believe this shouldn't happen", element.GetType(), element.UniqueId, topLevelMeshesCount);
```
Comment (jozseflkiss) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs:194]: {numSolids} --> {numMeshes}

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs [hunk 3]
```diff
@@ -127,6 +85,169 @@ void SortGeometry(GeometryElement geom, Transform inverseTransform = null)
...
+        }
+        if (topLevelGeomElementCount > 0)
+        {
+          SpeckleLog.Logger.Warning("Element of type {elementType} with uniqueId {uniqueId} has valid symbol geometry and {numSolids} top level geometry elements. See comment on method SortInstanceGeometry for link to RevitAPI docs that leads us to believe this shouldn't happen", element.GetType(), element.UniqueId, topLevelGeomElementCount);
```
Comment (jozseflkiss) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs:198]: {numSolids} --> {numGeomElements}

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs [hunk 4]
```diff
@@ -127,6 +85,169 @@ void SortGeometry(GeometryElement geom, Transform inverseTransform = null)
...
+        switch (geomObj)
+        {
+          case Solid:
+            return true;
+          case DB.Mesh:
+            return true;
+          case GeometryElement el:
+            return SymbolGeometryContainsGeometryData(el);
+          case GeometryInstance instance:
+            return SymbolGeometryContainsGeometryData(instance.GetSymbolGeometry());
```
Comment (jozseflkiss) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMeshUtils.cs:167]: instance is always not null here?

**Review by jozseflkiss — COMMENTED** [review]
I'm not so familiar with this code part, seems to be OK for me. Some minor suggestions added.

**Review by clairekuang — APPROVED** [review]

**Pull request merged**